classes/announcement/Announcement.php

```diff
@@ -35,92 +35,76 @@ class Announcement extends \PKP\core\DataObject
     //
     /**
      * Get assoc ID for this announcement.
-     *
-     * @return int
      */
-    public function getAssocId()
+    public function getAssocId(): ?int
     {
         return $this->getData('assocId');
     }
 
     /**
      * Set assoc ID for this announcement.
-     *
-     * @param int $assocId
      */
-    public function setAssocId($assocId)
+    public function setAssocId(?int $assocId): void
     {
         $this->setData('assocId', $assocId);
     }
 
     /**
      * Get assoc type for this announcement.
-     *
-     * @return int
      */
-    public function getAssocType()
+    public function getAssocType(): ?int
     {
         return $this->getData('assocType');
     }
 
     /**
      * Set assoc type for this announcement.
-     *
-     * @param int $assocType
      */
-    public function setAssocType($assocType)
+    public function setAssocType(?int $assocType): void
     {
         $this->setData('assocType', $assocType);
     }
 
     /**
      * Get the announcement type of the announcement.
-     *
-     * @return int
      */
-    public function getTypeId()
+    public function getTypeId(): ?int
     {
         return $this->getData('typeId');
     }
 
     /**
      * Set the announcement type of the announcement.
-     *
-     * @param int $typeId
      */
-    public function setTypeId($typeId)
+    public function setTypeId(?int $typeId): void
     {
         $this->setData('typeId', $typeId);
     }
 
     /**
      * Get the announcement type name of the announcement.
-     *
-     * @return string|null
      */
-    public function getAnnouncementTypeName()
+    public function getAnnouncementTypeName(): ?string
     {
+        if (!$this->getData('typeId')) {
+            return null;
+        }
         $announcementTypeDao = DAORegistry::getDAO('AnnouncementTypeDAO'); /** @var AnnouncementTypeDAO $announcementTypeDao */
-        $announcementType = $announcementTypeDao->getById($this->getData('typeId'));
-        return $announcementType ? $announcementType->getLocalizedTypeName() : null;
+        return $announcementTypeDao->getById($this->getData('typeId'))->getLocalizedTypeName();
```

Comment:
`$announcementTypeDao->getById($this->getData('typeId'))` could theoretically be null.

Reply:
Yep! I remember about changing this, it should be `?->`

Reply:
I just saw there's a `if (!$this->getData('typeId'))` before this line, so it should be ok for our standards.

Anyway, I'll add the extra check (for the theoretical case where the type is removed a couple of milliseconds after the announcement is retrieved).